src/renderer/modules/webpack.ts

```diff
@@ -137,193 +144,149 @@ function getExports(m: RawModule): ModuleExports | undefined {
-  if (typeof wpRequire?.c === "undefined") return;
-  return Object.values(wpRequire.c).find(filter);
+export function getExportsForProps<P extends string>(
+  m: ModuleExports,
+  props: P[],
+): ModuleExportsWithProps<P> | undefined {
+  if (typeof m !== "object") return;
+
+  return [m, ...Object.values(m)].find(
+    (o) =>
+      (typeof o === "object" || typeof o === "function") &&
+      o !== null &&
+      props.every((p) => p in o),
+  ) as ModuleExportsWithProps<P> | undefined;
 }
 
-export function getModule(filter: Filter): ModuleExports | undefined {
-  const raw = getRawModule(filter);
-  if (typeof raw !== "undefined") {
-    return getExports(raw);
-  }
-}
+export function getById(id: number, raw?: false): ModuleExports | undefined;
+export function getById(id: number, raw?: true): RawModule | undefined;
 
-export function getAllRawModules(filter?: Filter): RawModule[] {
-  const unfiltered = Object.values(wpRequire.c);
-  if (filter) {
-    return unfiltered.filter(filter);
+export function getById(id: number, raw = false): RawModule | ModuleExports | undefined {
+  if (!(id in wpRequire.c)) {
+    wpRequire(id);
   }
-  return unfiltered;
-}
 
-export function getAllModules(filter?: Filter): Array<ModuleExports | undefined> {
-  return getAllRawModules(filter).map(getExports);
-}
+  const rawModule: RawModule | undefined = wpRequire.c[id];
 
-function getExportsForProps<P extends string>(
-  m: RawModule,
-  props: P[],
-): ModuleExportsWithProps<P> | undefined {
-  if (typeof m.exports === "object") {
-    return [m.exports, ...Object.values(m.exports)].find(
-      (o) => typeof o === "object" && o !== null && props.every((p) => p in o),
-    ) as ModuleExportsWithProps<P> | undefined;
-  }
-}
-function byPropsInternal<P extends string>(props: P[], all: true): Array<ModuleExportsWithProps<P>>;
-function byPropsInternal<P extends string>(
-  props: P[],
-  all?: false,
-): ModuleExportsWithProps<P> | undefined;
-function byPropsInternal<P extends string>(
-  props: P[],
-  all = false,
-): (ModuleExportsWithProps<P> | undefined) | Array<ModuleExportsWithProps<P>> {
-  const result = [];
-  for (const m of getAllRawModules()) {
-    const exp = getExportsForProps(m, props);
-    if (exp) {
-      if (all) {
-        result.push(exp);
-      } else {
-        return exp;
-      }
-    }
-  }
-  if (all) {
-    return result;
+  if (raw) {
+    return rawModule;
   }
-}
 
-export function byPropsFilter<P extends string>(props: P[]) {
-  return (m: RawModule): m is RawModuleWithProps<P> => Boolean(getExportsForProps(m, props));
+  return typeof rawModule !== "undefined" ? getExports(rawModule) : void 0;
 }
 
-export function getByProps<P extends string>(...props: P[]): ModuleExportsWithProps<P> | undefined {
-  return byPropsInternal(props, false);
-}
+// Searcher
+
+// I'd prefer to use conditional types instead of overloading here, but I had some weird issues with it
```

Comment:
Elaborate?

Reply:
TypeScript doesn't have dependent typing; https://github.com/microsoft/TypeScript/issues/33014

Reply:
Can you add that to the comment?